llmfoundry/utils/warnings.py

```diff
@@ -38,3 +40,16 @@ def __init__(self, feature_name: str) -> None:
         super().__init__(
             f'{feature_name} is experimental and may change with future versions.'
         )
+
+
+# Decorator version of experimental warning
+def experimental(feature_name: str):
```

Comment:
nit: add doc string

Reply:
wdyt about making feature_name optional and defaulting to `func.__name__`?

Reply:
will add docstring

Reply:
on the default, i'd actually prefer to force the writer of the code to think about what the message will actually be, rather than default. not super strong opinion tho

Reply:
In theory, I agree with that, but in practice it seems like we'll just be writing the names of classes again and the classes _should_ just be named well. Would like to not have this extra room for misspellings haha

Reply:
well so far there are 3, and one of those i changed the string from the name to make more clear

Reply:
66% uses the default then 😄 ? tbh, don't feel strongly here, OK to leave as is